R/find_anchor_cells.R

```diff
@@ -64,11 +63,13 @@ get_anchor_stats <- function(counts, neg = NULL, bg = NULL, align_genes = TRUE,
   }
   
   # get cosine distances:
-  cos <- matrix(NA, nrow(counts), ncol(profiles),
-                dimnames = list(rownames(counts), colnames(profiles)))
-  cos <- sapply(colnames(profiles), function(cell) {
-    cos[, cell] <- apply(counts, 1, cosine, profiles[, cell])
-  })
+  cos_numerator <- counts %*% profiles
+  counts2 <- counts
+  counts2@x <- counts2@x^2
```

Comment:
`counts` might not be converted to a `dgCMatrix` if these functions are called outside of the `insitutype` semi-supervised method, see [semi-supervised vignette](https://github.com/Nanostring-Biostats/InSituType/blob/07ae239a0d4125327c00c5c30094f68f938a89ee/vignettes/NSCLC-semi-supervised-cell-typing-vignette.Rmd#L117) as example, in which case the `@x` part fails. Can you add a conversion with `as(counts, "dgCMatrix")`?

Reply:
@davidpross  I converted the counts to dgCMatrix in commit below.  let me know if any other changes or comments needed